contracts/cancun/cancun-opcodes/CancunOpcodes.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+contract CancunOpcodes {
+    event ContractAddress(address);
+
+    // @dev writes `value` to transient storage at `transientSlot` using tstore,
+    //      then read `value` from transient storage, using tload, into memory variable, val,
+    //      and finally write `val` to regular storage at `regularSlot`
+    function transientStorage(uint256 value, uint256 transientSlot, uint256 regularSlot) external {
+        // store `value` to `slot` using tstore
+        assembly {
+            tstore(transientSlot, value)
+        }
+
+        // The `value` at `transientSlot` is stored transiently and will be wiped off after the transaction is done processing.
+        // Therefore, in order to pertain the value for the sake of testing, value will be retrieved from the same transientSlot using TLOAD 
```

Comment:
"pertain" is probably not the word you were looking for?

Reply:
Ah I was confused with the word retain. Great catch thanks!